platform/core.network/src/org/netbeans/core/network/proxy/pac/impl/NbPacScriptEvaluator.java

```diff
@@ -336,41 +333,53 @@ private PacScriptEngine getScriptEngine(String pacSource) throws PacParsingExcep
 
             // Do some minimal testing of the validity of the PAC Script.
             final PacJsEntryFunction jsMainFunction;
-            if (nashornJava8u40Available) {
-                jsMainFunction = testScriptEngine(engine, true);
-            } else {
-                jsMainFunction = testScriptEngine(engine, false);
-            }
-            
+            jsMainFunction = testScriptEngine(engine, false);
+
             return new PacScriptEngine(engine, jsMainFunction);
         } catch (ScriptException ex) {
             throw new  PacParsingException(ex);
         }
     }
-    
-    private boolean getNashornJava8u40Available() {
-        try {
-            Class<?> klass = Class.forName("jdk.nashorn.api.scripting.NashornScriptEngineFactory");
-        } catch (ClassNotFoundException ex) {
-            return false;
-        }
-        return true;
+
+    @NbBundle.Messages({
+        "ALLOWED_PAC_ENGINES=GraalVM:js,Graal.js,Nashorn"
+    })
+    private static ScriptEngine newAllowedPacEngine(StringBuilder err) {
+        return newAllowedPacEngine(null, err);
     }
-    
-    private ScriptEngine getNashornJSScriptEngine() {
-        NashornScriptEngineFactory factory = new NashornScriptEngineFactory();
-        return factory.getScriptEngine(new ClassFilterPacHelpers());
+
+    public static ScriptEngine newAllowedPacEngine(String allowedEngines, StringBuilder err) {
+        if (allowedEngines == null) {
+            allowedEngines = Bundle.ALLOWED_PAC_ENGINES();
+        }
+        ScriptEngineManager manager = Scripting.newBuilder().build();
+        if (allowedEngines != null) {
```

Comment:
Can this be ever `null` - Bundle above would throw if key is not defined.

Reply:
Good catch. Result of additional refactorings: abd99c963a30b